process-test-results/results-transformer/transform.xml

```diff
@@ -4,16 +4,148 @@
 <xsl:stylesheet version="1.0"
     xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
     <xsl:output method="xml" indent="yes" encoding="UTF-8"/>
+
+    <!-- every testcase will have it's own, wrapping testsuite put around it -->
+    <xsl:template match="testsuites">
+      <testsuites>
+        <xsl:apply-templates select="testsuite/testcase"/>
+      </testsuites>
+    </xsl:template>
+
+    <!-- construct the testcase and wrapping testsuite for each testcase -->
     <xsl:template match="testcase">
       <testsuite name="{../@name}" tests="1" disabled="0" errors="0" failures="0" timestamp="{@timestamp}" time="{@time}">
-          <testcase name="{@name}" classname="{@classname}" time="{@time}">
-             <xsl:copy-of select="node()"/>
-          </testcase>
+        <testcase name="{@name}" classname="{@classname}" time="{@time}">
+          <xsl:apply-templates select="node()"/>
+        </testcase>
       </testsuite>
     </xsl:template>
-    <xsl:template match="testsuites">
-        <testsuites>
-            <xsl:apply-templates select="node()|@*"/>
-        </testsuites>
+
+    <!-- turn each flakyFailure in to a rerunFailure and turn system-out/err in to a text node -->
+    <xsl:template match="flakyFailure">
```

Comment:
I'm not familiar with xml, but the sight of this looks real good :)

Reply:
yeah xslt builds on xpath to reference components and applies transformations to the input xml.   Eventually this should not be needed and nextest should be updated, but first I needed to figure out what would work in allure, and xslt was the best way to learn that quickly